core/src/spotify_id.rs

```diff
@@ -257,29 +258,34 @@ impl SpotifyId {
     /// be encoded as `unknown`.
     ///
     /// [Spotify URI]: https://developer.spotify.com/documentation/web-api/#spotify-uris-and-ids
-    pub fn to_uri(&self) -> String {
+
+    #[allow(clippy::wrong_self_convention)]
+    pub fn to_uri(&self) -> Result<String, Error> {
         // 8 chars for the "spotify:" prefix + 1 colon + 22 chars base62 encoded ID  = 31
         // + unknown size item_type.
         let item_type: &str = self.item_type.into();
         let mut dst = String::with_capacity(31 + item_type.len());
         dst.push_str("spotify:");
         dst.push_str(item_type);
         dst.push(':');
-        dst.push_str(&self.to_base62());
+        let base_62 = self.to_base62()?;
+        dst.push_str(&base_62);
 
-        dst
+        Ok(dst)
     }
 }
 
 impl fmt::Debug for SpotifyId {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.debug_tuple("SpotifyId").field(&self.to_uri()).finish()
+        f.debug_tuple("SpotifyId")
+            .field(&self.to_uri().unwrap_or_default())
```

Comment:
In these debug methods, we may want to print something like "invalid uri" instead of defaulting to what probably is an empty string?

Reply:
Done.